### PR TITLE
[On-chain] feat: add `proof_window_open_offset_blocks` as `shared` module parameter

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Adds `proof_window_open_offset_blocks` parameter to the `shared` module

## Issue

- #516

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [x] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: x/shared/keeper/msg_server_update_param_test.go [hunk 1]
```diff
@@ -96,3 +96,31 @@ func TestMsgUpdateParam_UpdateClaimWindowCloseOffsetBlocks(t *testing.T) {
...
+	msgSrv := keeper.NewMsgServerImpl(k)
+
+	// Set the parameters to their default values
+	defaultParams := sharedtypes.DefaultParams()
+	require.NoError(t, k.SetParams(ctx, defaultParams))
+
+	// Ensure the default values are different from the new values we want to set
+	require.NotEqual(t, uint64(expectedProofWindowOpenOffestBlocks), defaultParams.ProofWindowOpenOffsetBlocks)
+
+	// Update the claim window close offset blocks param
```
Comment (bryanchriswhite) [inline x/shared/keeper/msg_server_update_param_test.go:113]:
```suggestion
	// Update the proof window open offset blocks param
```

**Review by red-0ne — COMMENTED** [review]
Missing a `message_update_param` and makefile target but otherwise, LGTM

**Review by Olshansk — COMMENTED** [review]
Not a request but a question:

1. Why no Makefile target?
2. Do we have a unit test for failing edge cases elsewhere? If so, we should have it here too

**Comment by bryanchriswhite** [comment]
> Not a request but a question:
> 
> 1. Why no Makefile target?

Added in [fe30755](https://github.com/pokt-network/poktroll/pull/569/commits/fe307553c7debaaf1e79f77b0cde8e3d2793071f).

> 2. Do we have a unit test for failing edge cases elsewhere? If so, we should have it here too

@Olshansk can you elaborate on what you mean here?

The first thing that `msgServer#UpdateParams()` does is call `MsgUpdateParam#ValdiateBasic()`, which has almost complete coverage:

image
> Doesn't cover `MsgUpdateParam#AsType == nil` case.

The rest of the un-covered (sad) paths in `msgServer#UpdateParam()` are redundant as they would only occur if `MsgUpdateParam#ValidateBasic()` returned an error, which we're checking for initially:

image
> Doesn't cover redundant sad paths.

**Review by Olshansk — APPROVED** [review]
@bryanchriswhite Appreciate the explanation in https://github.com/pokt-network/poktroll/pull/569#issuecomment-2144820855.

One small comment (related to comments in the Makefile), but otherwise LGTM.
